doc/releases.md

```diff
@@ -2,6 +2,24 @@
 
 See [the HoloViz blog](https://blog.holoviz.org/tag/panel.html) for a visual summary of the major features added in each release.
 
+## Version 0.14.4
+
+Date: 2023-03-02
+
+This release is a small bug fix release preceding the upcoming major release of Panel 1.0. Many thanks to the contributors to this release which include @MarcSkovMadsen, @maximlt, @Hoxbro and @philippjfr.
+
+### Bugs
+
+- Client-side search filters don't use regex ([4423](https://github.com/holoviz/panel/pull/4423))
```

Comment:
We (or rather I) haven't generally been great about this but it would be good to summarize what this fixes rather than the technical details of the fix, i.e. make it something meaningful for a user.

Reply:
Indeed, I always find that a little tricky to phrase correctly, will try to do a better job at it!